espnet2/samplers/batch_bin_sampler.py

```diff
@@ -0,0 +1,153 @@
+from typing import Iterator
+from typing import List
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+from typeguard import check_argument_types
+
+from espnet2.samplers.abs_sampler import AbsSampler
+from espnet2.utils.fileio import load_num_sequence_text
+
+
+class BatchBinSampler(AbsSampler):
+    def __init__(
+        self,
+        batch_bins: int,
+        shape_files: Union[Tuple[str, ...], List[str]],
```

Comment:
Maybe `typing.Sequence[str]` is better?

Reply:
(You don't need to check typing though there are some miss, that is not related to this PR)

I basically don't use Sequence[str].
str is also `Sequence[str]`, so the desired type is actually `not str and Sequence[str]`, but this looks a little bit ugly, so.

Reply:
It looks crazy. I hate it too indeed.